packages/installer/src/dappGet/fetch/DappGetFetcher.ts

```diff
@@ -78,7 +68,93 @@ export class DappGetFetcher {
       }
     }
-    // Case 3. unvalid semver version ("/ipfs/Qmre4..."): Asume it's the only version
+    // Case 3. invalid semver version ("/ipfs/Qmre4..."): Assume it's the only version
     return [versionRange];
   }
+
+  /**
+   * Merges optional dependencies into the main dependencies object if the corresponding 
+   * packages are installed.
+   *
+   * @param dependencies The main dependencies object to be merged into.
+   * @param optionalDependencies The optional dependencies to be checked and merged.
+   * @param installedPackages The list of currently installed packages.
+   */
+  private mergeOptionalDependencies(
+    dependencies: Dependencies,
+    optionalDependencies: Dependencies,
+    installedPackages: InstalledPackageData[]
+  ): void {
+    for (const [optionalDepName, optionalDepVersion] of Object.entries(optionalDependencies)) {
+      const isInstalled = installedPackages.some(
+        (installedPackage) => installedPackage.dnpName === optionalDepName
+      );
+
+      if (isInstalled) {
+        dependencies[optionalDepName] = optionalDepVersion;
+      }
+    }
+  }
+
+  /**
+   * Processes the dependencies by first filtering out those that are already satisfied by installed packages
+   * and then converting any remaining semver ranges to appropriate APM versions.
+   *
+   * @param dependencies The main dependencies object to be processed.
+   * @param installedPackages The list of currently installed packages.
+   */
+  private processDependencies(
+    dependencies: Dependencies,
+    installedPackages: InstalledPackageData[]
+  ): void {
+    for (const [depName, depVersion] of Object.entries(dependencies)) {
+      const installedPackage = installedPackages.find(
+        (pkg) => pkg.dnpName === depName
+      );
+
+      if (!validRange(depVersion))
+        throw new Error(`Invalid semver notation for dependency ${depName}: ${depVersion}`);
+
+      if (depVersion.includes('||') || depVersion.includes(' ')) {
+        throw new Error(`Unsupported version range for dependency ${depName}: ${depVersion}. Only simple ranges are supported`);
+      }
+
+      // Remove dependency if it is already satisfied by an installed package
+      if (installedPackage && satisfies(installedPackage.version, depVersion)) {
+        console.log(
+          `Dependency ${depName} is already installed with version ${installedPackage.version}`
+        );
+        delete dependencies[depName];
+        continue;
+      }
+
+      dependencies[depName] = this.parseSemverRangeToApmVersion(depVersion);
+    }
+  }
+
+  /**
+   * Parses a semver range and converts it to a version suitable for APM.
+   * Handles specific semver operators like ^, ~, >, >= and converts them to
+   * appropriate APM-compatible versions.
+   *
+   * @param semverRange The semver range to be parsed.
+   * @returns The parsed version as a string.
+   */
+  private parseSemverRangeToApmVersion(semverRange: string): string {
```

Comment:
I would rename the function to something like `parseSemverRangeToSemverStrict` since semver is something not specific to APM

Reply:
This function has been removed, as it is not needed anymore